packages/botonic-api/src/rest/routes/events.ts

```diff
@@ -10,169 +18,210 @@ import {
   validateBotonicEventData,
 } from './validation/events'
 
-const router = Router()
-
-router
-  .route('/')
-  .get(
-    checkSchema({
-      page: pageParamSchema,
-      pageSize: pageSizeParamSchema,
-    }),
-    async (req, res) => {
-      try {
-        const errors = validationResult(req)
-        if (!errors.isEmpty()) {
-          res.status(400).send({ errors: errors.array() })
-          return
+export default function eventsRouter(args: any): Router {
+  const router = Router()
+  const botDispatcher: BotDispatcher = args.botDispatcher
+  router
+    .route('/')
+    .get(
+      checkSchema({
+        page: pageParamSchema,
+        pageSize: pageSizeParamSchema,
+      }),
+      async (req, res) => {
+        try {
+          const errors = validationResult(req)
+          if (!errors.isEmpty()) {
+            res.status(400).send({ errors: errors.array() })
+            return
+          }
+
+          const query = matchedData(req, { locations: ['query'] })
+          const paginator = new Paginator(req, query.page, query.pageSize)
+          const dp = dataProviderFactory(process.env.DATA_PROVIDER_URL)
+          const events = await dp.getEvents(paginator.limit, paginator.offset)
+
+          const response = paginator.generateResponse(events)
+          res.status(200).json(response)
+        } catch (e) {
+          res.status(500).send({ error: e.message })
         }
-
-        const query = matchedData(req, { locations: ['query'] })
-        const paginator = new Paginator(req, query.page, query.pageSize)
+      }
+    )
+    // // Current route
+    // .post(async (req, res) => {
+    //   const errors = await validateBotonicEventData({ request: req })
+    //   if (!errors.isEmpty()) {
+    //     res.status(400).send({ errors: errors.array({ onlyFirstError: true }) })
+    //     return
+    //   }
+
+    //   try {
+    //     const event = matchedData(req, { locations: ['body'] }) as BotonicEvent
+    //     const dp = dataProviderFactory(process.env.DATA_PROVIDER_URL)
+    //     if (await dp.getEvent(event.eventId)) {
+    //       res
+    //         .status(409)
+    //         .send({ error: `Event with ID '${event.eventId} already exists` })
+    //     }
+    //     const storedEvent = await dp.saveEvent(event)
+    //     res.status(201).send(storedEvent)
+    //   } catch (e) {
+    //     res.status(500).send({ error: e.message })
+    //   }
+    // })
+    .post(
+      jwt({ secret: process.env.BOTONIC_JWT_SECRET, algorithms: ['HS256'] }),
```

Comment:
Refactored: https://github.com/hubtype/botonic/pull/1834/commits/c6c5d2bd3556928626ae24d557b2deef1a10299b
Regarding adding it to every endpoint, we need to discuss it. The luck here is that I think it will be easier to do it with `router.use` with this jwt handler. Wrote it down for next iterations